### ISSUE TITLE
Package updates for 2024

### ISSUE THREAD

**Issue by weiji14** [CryoInTheCloud/hub-image#102]
TODO list of packages to add/update in 2024.

High priority:

- [x] :arrow_up: QGIS 3.28 Firenze to [QGIS 3.34 Prizren](https://qgis.org/en/site/forusers/visualchangelog334/index.html)
- [x] :arrow_up: JupyterLab v3 to v4
- [x] :arrow_up: jupyterlab_myst to v2.1.0
- [x] :heavy_plus_sign: VScode
- [x] :heavy_plus_sign: https://github.com/silx-kit/jupyterlab-h5web

Low priority:
- [ ] :arrow_up: Python 3.11 to 3.12
- [x] :heavy_plus_sign: [jupyter-ai](https://jupyter-ai.readthedocs.io/en/latest/) - Jupyter AI 1.x supports JupyterLab 3.x, and Jupyter AI 2.x supports JupyterLab 4.x

Feel free anyone to request package updates by commenting below, or opening a Pull Request directly following instructions at https://github.com/CryoInTheCloud/hub-image#updating-packages-in-this-repository

---

Previous package update issue is at #86.

**Comment by tsnow03** [comment]
Thanks @weiji14 for the bump on these! I added a MyST update and adding VScode.

**Comment by tsnow03** [comment]
Also adding the jupyter-ai extension

**Comment by betolink** [comment]
Hi @weiji14 and @tsnow03 I think we need to update earthaccess to `v0.9.0` and install voila to notebooks in the form of widgets, should I open a PR to modify the environment file?

**Comment by weiji14** [comment]
> Hi @weiji14 and @tsnow03 I think we need to update earthaccess to `v0.9.0` and install voila to serve notebooks like widgets, should I open a PR to modify the environment file?

Yes please, go ahead and open a PR to modify the line here: https://github.com/CryoInTheCloud/hub-image/blob/a0063519a32a280fc6ba1dd9a6a3b402e7f53299/environment.yml#L100

**Comment by weiji14** [comment]
Closing for the 2025 ticket - https://github.com/CryoInTheCloud/hub-image/issues/129
